packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart

```diff
@@ -0,0 +1,204 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/command_runner.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/create.dart';
+import 'package:flutter_tools/src/dart/pub.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../src/common.dart';
+import '../src/context.dart';
+import '../src/testbed.dart';
+
+void main() {
+  Directory tempDir;
+  Directory projectDir;
+
+  setUpAll(() async {
+    Cache.disableLocking();
+    await _ensureFlutterToolsSnapshot();
+  });
+
+  setUp(() {
+    tempDir = globals.fs.systemTempDirectory
+        .createTempSync('flutter_tools_generated_plugin_registrant_test.');
+    projectDir = tempDir.childDirectory('flutter_project');
+  });
+
+  tearDown(() {
+    tryToDelete(tempDir);
+  });
+
+  testUsingContext('generated plugin registrant passes analysis', () async {
+    await _createProject(projectDir, <String>[]);
+    // We need to add a dependency with web support to trigger
+    // the generated_plugin_registrant generation.
+    await _addDependency(projectDir, 'shared_preferences', '^0.5.12+4');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+
+  testUsingContext(
+      'generated plugin registrant ignores lines longer than 80 chars',
+      () async {
+    await _createProject(projectDir, <String>[]);
+    await _addAnalysisOptions(
+        projectDir, <String>['lines_longer_than_80_chars']);
+    await _createProject(tempDir.childDirectory('test_plugin'), <String>[
+      '--template=plugin',
+      '--platforms=web',
+      '--project-name',
+      'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+    ]);
+    // The line for the test web plugin (`  TestWebPluginWithAPurposefullyExtremelyLongPackageNameWeb.registerWith(registrar);`)
+    // exceeds 80 chars.
+    // With the above lint rule added, we want to ensure that the `generated_plugin_registrant.dart`
+    // file does not fail analysis (this is a regression test - an ignore was
+    // added to cover this case).
+    await _addDependency(
+        projectDir,
+        'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+        '\n        path: ../test_plugin');
```

Comment:
prefer putting the `\n` at the end of the previous line rather than the start of the new line

Reply:
The problem here is that I wanted to separate the package key and the version value - I will refactor this to allow for `path` properly.